common/lib/client/push.js

```diff
@@ -0,0 +1,134 @@
+var Push = (function() {
+	var msgpack = (typeof require !== 'function') ? Ably.msgpack : require('msgpack-js');
+
+	function Push(rest) {
+		this.rest = rest;
+		this.admin = new Admin(rest);
+	}
+
+	Push.prototype.publish = function(recipient, payload, callback) {
+		var rest = this.rest;
+		var format = rest.options.useBinaryProtocol ? 'msgpack' : 'json',
+		    requestBody = Utils.mixin({recipient: recipient}, payload),
+		    headers = Utils.copy(Utils.defaultPostHeaders(format));
+
+		if(rest.options.headers)
+			Utils.mixin(headers, rest.options.headers);
+
+		requestBody = (format == 'msgpack') ? msgpack.encode(requestBody, true): JSON.stringify(requestBody);
```

Comment:
Given that's all you're using the `format` variable for, you could get rid of it and just do `requestBody = rest.options.useBinaryProtocol ? msgpack.encode(requestBody, true): JSON.stringify(requestBody)`